hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/table/format/FormatUtils.java

```diff
@@ -208,19 +205,33 @@ public BoundedMemoryRecords(
         Schema logSchema,
         Configuration hadoopConf,
         org.apache.flink.configuration.Configuration flinkConf) {
+      HoodieUnMergedLogRecordScanner.Builder scannerBuilder = HoodieUnMergedLogRecordScanner.newBuilder()
```

Comment:
This refactoring is necessary to avoid exposing internals of the executor (queue), instead relying on the provided interfaces to configure it